kernel/arch/dreamcast/include/dc/perf_monitor.h

```diff
@@ -0,0 +1,96 @@
+/* KallistiOS ##version##
+
+   arch/dreamcast/include/dc/perf_monitor.h
+ * Copyright (C) 2024 Paul Cercueil
+
+*/
+
+/** \file    dc/perf_monitor.h
+    \brief   Low-level performance monitor
+    \ingroup perf_monitor
+
+    This file contains an API that can be used to monitor specific
+    performance events in one or several functional blocks.
+
+    \author Paul Cercueil
+*/
+
+#ifndef __KOS_PERF_MONITOR_H
+#define __KOS_PERF_MONITOR_H
+
+#include <sys/cdefs.h>
+__BEGIN_DECLS
+
+#include <dc/perfctr.h>
+#include <stdint.h>
+#include <stdio.h>
+
+/** \defgroup   perf_monitor Performance monitor
+    \brief      Code performance monitor
+    \ingroup    debugging
+
+    The performance monitor API is built on top of the performance counter API,
+    and as such cannot be used at the same time.
+    With this API, programs can set probe points in different functional blocks
+    and later obtain statistics about the execution of said functional blocks.
+
+    @{
+*/
+
+/** /cond */
+struct perf_monitor {
+	const char *fn;
+	unsigned int line;
+	uint64_t calls;
+	uint64_t time_ns, time_start;
+	uint64_t event0, event0_start;
+	uint64_t event1, event1_start;
+};
```

Comment:
```suggestion
} perf_monitor_t;
```
Sorry, required by law to add the typedef. ;)

Reply:
Wait, it's technically not a user-facing API structure... and you even marked it as such for the Doxygen... fine. I will accept the lack of tyepdefing. lol.